### DISCUSSION TITLE
Options for JSLint in VSCode

### DISCUSSION THREAD

**Discussion by erosman** [jslint-org/jslint#432] [General]
Is there a setting file for JSLint options in VSCode?

**Comment by kaizhu256** [comment]
- no, but admittedly, i'm not a regular VSCode user, so not well-versed with its use-cases
- extension `vscode-jslint` was designed to be:
  - zero-config
  - zero-dependency (aside from nodejs)
  - never auto-lint
- what file settings do you have in mind?
  - if it improves the overall ux, it might change my mind on the previous design criterias

**Comment by erosman** [comment]
There are warnings generated that are not desired.

I had to edit the `jslint.mjs` to disable those e.g.

```js
option_dict.single = true;
option_dict.long = true;
// ...
```

**Comment by kaizhu256** [comment]
- for directive-related config-settings, i generally hard-code them near top of each file, e.g.

```javascript
// file foo.mjs
/*jslint
    long,
    node,
    single,
    ....
*/

import fs from 'fs';
...
```

- -1 in terms of ux, for changing above convention from per-file-basis to config-basis
  - when re-using files from old project to new project
  - you additionally need to edit a config in order for jslint to properly lint the re-used file

**Comment by erosman** [comment]
In browser extension development, such in-file config is not desirable, neither is adding identical config to each and every file.
